plotting/auxiliars_withSF.py

```diff
@@ -0,0 +1,238 @@
+  unc = 0
+  if var != 0: # Uncertainties added in quadrature
+    unc  = (muID[year].GetBinError(min(max(1,muID[year].GetXaxis().FindBin(abs(eta))),maxmidx[year]), min(max(1,muID[year].GetYaxis().FindBin(pt)),maxmidy[year])))**2
+    unc += (muISO[year].GetBinError(min(max(1,muISO[year].GetXaxis().FindBin(abs(eta))),maxmisox[year]), min(max(1,muISO[year].GetYaxis().FindBin(pt)),maxmisoy[year])))**2
+    unc = unc**0.5
+  return sf + unc*var
+
+def getSFEl(pt, eta, year, var):
+  sf  = eID[year].GetBinContent(min(max(1,eID[year].GetXaxis().FindBin(eta)),maxeidx[year]), min(max(1,eID[year].GetYaxis().FindBin(pt)),maxeidy[year]))
+  sf *= eRECO[year].GetBinContent(min(max(1,eRECO[year].GetXaxis().FindBin(eta)),maxerex[year]), min(max(1,eRECO[year].GetYaxis().FindBin(pt)),maxerey[year]))
+  unc = 0
+  if var != 0: # Uncertainties added in quadrature
+    unc  = (eID[year].GetBinError(min(max(1,eID[year].GetXaxis().FindBin(eta)),maxeidx[year]), min(max(1,eID[year].GetYaxis().FindBin(pt)),maxeidy[year])))**2
+    unc += (eRECO[year].GetBinError(min(max(1,eRECO[year].GetXaxis().FindBin(eta)),maxerex[year]), min(max(1,eRECO[year].GetYaxis().FindBin(pt)),maxerey[year])))**2
+    unc = unc**0.5
+  return sf + unc*var
+
+def getSF2Mu(lep1pt, lep1eta, lep2pt, lep2eta, year, var, mode = 0):
+  ceval = CorrectionSet.from_file("C:/Users/slatt/Downloads/POGJsons/masterJSON.json") #Load JSON file with Trigger Efficiencies
+
+  if lep1pt > 200.0: #No Trigger Scale Factor Applied (Out of bounds Pt)
+    return getSFMu(lep1pt, lep1eta, year, var)*getSFMu(lep2pt, lep2eta, year, var)
+
+  if mode == 0: #Pt-Eta Binning for Trigger Scale Factors:
+    if var > 0:
+      if ceval['UL-PtEta-Trigger-SFs'].evaluate(year,"sf-up","Muon",lep1eta, lep1pt) < 0.25: #Make sure uncertainty in bin is not too high
+        return getSFMu(lep1pt, lep1eta, year, var)*getSFMu(lep2pt, lep2eta, year, var)*(ceval['UL-PtEta-Trigger-SFs'].evaluate(year,"sf","Muon",lep1eta, lep1pt) + var * ceval['UL-PtEta-Trigger-SFs'].evaluate(year,"sf-up","Muon",lep1eta, lep1pt))
+    else:
+      if ceval['UL-PtEta-Trigger-SFs'].evaluate(year,"sf-down","Muon",lep1eta, lep1pt) < 0.25: #Make sure uncertainty in bin is not too high
+        return getSFMu(lep1pt, lep1eta, year, var)*getSFMu(lep2pt, lep2eta, year, var)*(ceval['UL-PtEta-Trigger-SFs'].evaluate(year,"sf","Muon",lep1eta, lep1pt) + var * ceval['UL-PtEta-Trigger-SFs'].evaluate(year,"sf-down","Muon",lep1eta, lep1pt))
+
+  if mode == 1: #Pt-Pt Binning for Trigger Scale Factors:
+    if var > 0:
+      if ceval['UL-PtPt-Trigger-SFs'].evaluate(year,"sf-up","Muon",lep2pt, lep1pt) < 0.25: #Make sure uncertainty in bin is not too high
+        return getSFMu(lep1pt, lep1eta, year, var)*getSFMu(lep2pt, lep2eta, year, var)*(ceval['UL-PtPt-Trigger-SFs'].evaluate(year,"sf","Muon",lep2pt, lep1pt) + var * ceval['UL-PtPt-Trigger-SFs'].evaluate(year,"sf-up","Muon",lep2pt, lep1pt))
```

Comment:
Out of curiosity. Is the order correct? I.e. first the second lepton then the first?